File: challenges/duplicate_zeros.py


class Solution(object):
	"""
	"""
	def duplicateZeros(self, arg):
		arr_copy = arr[:]
		index, n = 0, len(arr_copy)
		for elem in arr_copy:
			arr[index] = elem
			index += 1
			if index >= n:
				break
			if elem == 0:
				arr[index] = elem
				index += 1
				if index >= n:
					break

File: challenges/duplicate-zeros/solutions/duplicate-zeros.js
function duplicateZeros(array) {
  let length = array.length;
  for (i = 0; i < array.length; i++) {
    if (array[i] === 0) {
      i++;
      array.splice(i - 1, 0, 0);
      array.length = length;
    }
  }
  //console.log(array)
}
